src/main/java/org/jabref/gui/search/GlobalSearchBar.java

```diff
@@ -217,9 +209,11 @@ public void performSearch() {
 
         Globals.stateManager.setSearchQuery(searchQuery);
 
-        // TODO: Remove search worker as this is doing the work twice now
-        searchWorker = new SearchWorker(currentBasePanel, searchQuery, searchDisplayMode);
-        searchWorker.execute();
+        updateResults(Globals.stateManager.searchResultSizeProperty().get(),
```

Comment:
Is this really working as desired? I would have expected that the `resultSize` is not yet updated correctly (for large databases).

Reply:
Good thing. I'm trying to bind it to the results property but turns out to be a bit more complicated as the illegal search query msg is also listed in the label